internal/irc/handler.go

```diff
@@ -260,7 +295,29 @@ func (h *Handler) Run() error {
 		return err
 	}
 
-	h.client.Loop()
+	shouldDisconnect := false
+	h.m.Lock()
+	switch h.clientState {
+	case ircStopped:
+		// concurrent Stop(), bail
+		shouldDisconnect = true
+	case ircConnecting:
+		// success!
+		h.client = client
+		h.clientState = ircLive
+	case ircLive:
+		// impossible
+		h.log.Error().Stack().Msgf("two concurrent connection attempts detected")
+		shouldDisconnect = true
+	}
+	h.m.Unlock()
```

Comment:
This fixes the TOCTOU race identified by @frrad previously: if `Stop()` is called while we were trying to connect, just tear down the connection instead of saving it.